var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -910,10 +929,16 @@ def define_tpl(trilinos_name, spack_name, have_dep):
                         define_kok_enable("HIP_RELOCATABLE_DEVICE_CODE", "rocm_rdc"),
                     ]
                 )
+                # The following is an optimization to compile only for the specified
+                # archs since all archs are done by default
+                amd_targets = self.spec.variants["amdgpu_target"].value
+                options.append("-DCMAKE_HIP_ARCHITECTURES=" + ";".join(str(x) for x in targets))
```

Comment:
`self.define` automatically formats lists.
```suggestion
                options.append(self.define("CMAKE_HIP_ARCHITECTURES", targets))
```

Also how are this and the AMDGPU/GPU defines (and the kokkos defines) related?

Reply:
This is separate from Kokkos. These override stuff in `find_package(HIP)` and I've found it necessary to list them all. Talking with someone at AMD, it should just be one option in the future. However, this isn't strictly necessary, just builds faster.